.github/workflows/trailing-commas.py

```diff
@@ -0,0 +1,66 @@
+"""This is an pretiffy script."""
+import os
+import sys
+import argparse
+
+
+def arg_parser_resolver():
+    """Parse the files of the directory."""
+    parser = argparse.ArgumentParser(
+        description='for parsing across the directory')
+    parser.add_argument('--dir', type=str, required=False, default=os.getcwd(),
+                        help='directory-location where files are present')
+    return parser.parse_args()
+
+
+def syntax_matcher(root: str, files: list):
+    """Patterm matching and adding trailing commas at the required places."""
+    for name in files:
+
+        file_location = os.path.join(root, name)
+        data = []
+        read_loc = open(file_location, 'r')
+        data = read_loc.readlines()
+        write_loc = open(file_location, 'w')
+
+        for index in range(0, len(data)):
+            # genral case of addition of commas
+            if '))' and not ')) {' in data[index]:
+                data[index] = data[index].replace('))', '),)')
+
+            # for cases of function declaration without any parameters
+            if '(,)' in data[index]:
+                data[index] = data[index].replace('(,)', '()')
+
+            # for commas already exist and formatting is already done
+            if ',));' in data[index]:
+                data[index] = data[index].replace(',));', '));')
+
+            # for the unique eception that occured in grops_controller.dart
+            if name == 'groups_controller.dart':
```

Comment:
Will this type of changes will be needed in future?
Like if a contributor is creating a controller or other file, then do they need to make changes to this script too? If yes, then they need to have at least a very basic understanding of python.

Reply:
I am thinking of a sustainable method to inculcate this into a logical condition. But it is taking down 10+ modifications. I will work it through